mapper.cpp

```diff
@@ -56,11 +86,22 @@ void Mapper::align_mate_in_window(Alignment& read1, Alignment& read2, int pair_w
     int64_t first = max((int64_t)0, idf - pair_window);
     int64_t last = idl + (int64_t) pair_window;
     VG* graph = new VG;
-    index->get_range(first, last, *graph);
     
-    // We also want to get nodes connected to these nodes along edges, down to the specified depth.
-    // This is useful for graphs that have unhelpful node ID assignments.
-    index->expand_context(*graph, context_depth);
+    // Now we need to get the neighborhood by ID and expand outward by actual
+    // edges. How we do this depends on what indexing structures we have.
+    if(xindex) {
+        Graph new_stuff;
```

Comment:
Note that there is the function [XG::neighborhood](https://github.com/ekg/xg/blob/3c7002cf707af09f999b7e1dea7a6cacfec8c8dc/xg.cpp#L923). This looks fine though.